### PR TITLE
#3095 #3161 #3163

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#3095 Function arguments in tooltips messed up 
- this was techically by design since 0.3 but apparently wrong. I thought that among function aliases signature would be the same but it is not true. Alias means more 'function from the same family'. We now match RS.

#3161 Completion shows signature from wrong function
- make sure we extract package name and use it when fetching function information. This used to work when we had functions extracted from all packages but stopped when we fixed behaviour so only function from the last loaded package is shown. Test TBA, need to figure out stability since tests with package load/unload are not very stable.

#3163 Dismissing signature help with ESC adds line break in REPL
- check that there is no completion or signature sessions up

### CODE REVIEW

File: src/R/Editor/Impl/Signatures/ParametersInfo.cs [hunk 1]
```diff
@@ -35,19 +35,25 @@ public sealed class ParameterInfo {
         public string FunctionName { get; private set; }
 
         /// <summary>
+        /// Function name
```
Comment (AlexanderSher) [inline src/R/Editor/Impl/Signatures/ParametersInfo.cs]: Wrong comment?

File: src/Package/Impl/Repl/Commands/ReplCommandController.cs [hunk 2]
```diff
@@ -152,10 +152,12 @@ public ReplCommandController(ITextView textView, ITextBuffer textBuffer)
             }
         }
 
-        private void HandleCancel(RCompletionController controller) {
-            Workflow.Operations.CancelAsync().DoNotWait();
-            // Post interrupt command which knows if it can interrupt R or not
-            VsAppShell.Current.PostCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdInterruptR);
+        private void HandleCancel(RCompletionController controller, ITextView textView) {
```
Comment (AlexanderSher) [inline src/Package/Impl/Repl/Commands/ReplCommandController.cs]: `TextView` is a field of the same class. Did you want to make this method static?

File: src/R/Editor/Impl/Signatures/ParametersInfo.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Editor/Impl/Signatures/ParametersInfo.cs]: Yes

File: src/Package/Impl/Repl/Commands/ReplCommandController.cs (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/Repl/Commands/ReplCommandController.cs]: I'll use the field instead

File: src/Package/Impl/Repl/Commands/ReplCommandController.cs [hunk 3]
```diff
@@ -152,10 +152,12 @@ public ReplCommandController(ITextView textView, ITextBuffer textBuffer)
...
+            if (!controller.HasActiveCompletionSession && !controller.HasActiveSignatureSession(textView)) {
```
Comment (AlexanderSher) [inline src/Package/Impl/Repl/Commands/ReplCommandController.cs]: I don't understand this. `InterruptRCommand` command calls `_session.CancelAllAsync`, which is also called from `Workflow.Operations.CancelAsync()`. Why do we call it twice?

Comment (MikhailArkhipov) [inline src/Package/Impl/Repl/Commands/ReplCommandController.cs]: InterruptR may or may not work. Example is when typing incomplete expression user wants out and hits ESC. It should return user to `>`. However, InterruptR doesnt work for that. Now, if there is a long running process then InterruptR works (and clears pending inputs). Also, InterruptR is disabled when in debugger break mode but ESC should still work.

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**